app-modules/inventory-management/src/Models/AssetCheckOut.php

```diff
@@ -116,4 +118,19 @@ public static function getTimelineData(Model $forModel): Collection
     {
         return $forModel->checkOuts()->get();
     }
+
+    protected function status(): Attribute
+    {
+        return Attribute::get(function () {
+            if (($this->checkIn()->exists())) {
+                return AssetCheckOutStatus::Returned;
+            }
+
+            if ($this->expected_check_in_at->isPast()) {
+                return AssetCheckOutStatus::PastDue;
+            }
+
+            return AssetCheckOutStatus::InGoodStanding;
+        });
```

Comment:
Do you think this could pose a problem in the future if we needed to allow for filtering based on this status?

We don't necessarily need to change it now, but I have some ideas if we need to in the future.

Reply:
I don't think so, but we can port this onto the table at some point if it does become tricky, or needs to be a little more robust than this.